src/main/java/edu/kis/powp/jobs2d/drivers/adapter/controller/DeviceControllerDriver.java

```diff
@@ -0,0 +1,66 @@
+package edu.kis.powp.jobs2d.drivers.adapter.controller;
+
+import edu.kis.powp.jobs2d.drivers.VisitableJob2dDriver;
+import edu.kis.powp.jobs2d.drivers.adapter.DriverDecorator;
+import edu.kis.powp.jobs2d.drivers.singleton.DeviceController;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class DeviceControllerDriver implements DriverDecorator {
+
+    private VisitableJob2dDriver driver;
+    private final List<IThresholdListener> listeners = new ArrayList<>();
```

Comment:
OK, but the observer pattern is already implemented in this project (Publisher & Subscriber) classes.